### PR TITLE
Refactor integration tests

### PR DESCRIPTION
**Pull request by markcty**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
To prevent the coverage from fluctuating.

* what changes does this pull request make?
Move all IT-like UT tests to IT tests.

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
No.

### CODE REVIEW

File: curp/src/rpc/connect.rs [hunk 1]
```diff
@@ -127,9 +154,10 @@ impl ConnectInterface for Connect {
         request: ProposeRequest,
         timeout: Duration,
     ) -> Result<tonic::Response<ProposeResponse>, ProposeError> {
-        #[cfg(test)]
-        if !self.reachable.load(Ordering::Relaxed) {
-            return Err(ProposeError::RpcError("unreachable".to_owned()));
+        if let Some(ref f) = self.tx_filter {
```
Comment (Phoenix500526) [inline curp/src/rpc/connect.rs:157]: We can pack this code as a method `check_filter` or something like that. And call it here to reduce some duplicated code

File: curp/tests/server_election.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+async fn reelect() {
+    init_logger();
+
+    let group = CurpGroup::new(5).await;
+
+    // check whether there is exact one leader in the group
+    let leader1 = group.get_leader().await.0;
+    let term1 = group.get_term_checked().await;
+
+    ////////// disable leader 1
```
Comment (Phoenix500526) [inline curp/tests/server_election.rs:55]: Using '//' is enough.

File: curp/tests/server_election.rs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    println!("disable leader {leader1}");
+
+    // after some time, a new leader should be elected
+    tokio::time::sleep(Duration::from_secs(2)).await;
+    let (leader2, term2) = group.get_leader().await;
+
+    assert_ne!(term1, term2);
+    assert_ne!(leader1, leader2);
+
+    ////////// disable leader 2
```
Comment (Phoenix500526) [inline curp/tests/server_election.rs:66]: Same above

File: curp/tests/server_election.rs [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+    ////////// disable leader 3
+    group.disable_node(&leader3);
+    println!("disable leader {leader3}");
+
+    // after some time, no leader should be elected
+    tokio::time::sleep(Duration::from_secs(2)).await;
+    assert!(group.try_get_leader().await.is_none());
+
+    ////////// recover network partition
```
Comment (Phoenix500526) [inline curp/tests/server_election.rs:87]: Same above

File: curp/tests/server_recovery.rs [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+    let req1 = ProposeRequest {
+        command: bincode::serialize(&cmd1).unwrap(),
+    };
+    for id in group.all.keys().filter(|&id| id != &leader1).take(4) {
+        let mut connect = group.get_connect(id).await;
+        connect.propose(req1.clone()).await.unwrap();
+    }
+    tokio::time::sleep(Duration::from_secs(1)).await;
+
+    // 2
```
Comment (Phoenix500526) [inline curp/tests/server_recovery.rs:186]: Is here missing something?

File: curp/tests/server_recovery.rs [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+#[tokio::test]
+async fn new_leader_will_recover_spec_cmds_cond2() {
+    init_logger();
+
+    let group = CurpGroup::new(5).await;
+    let client = group.new_client(ClientTimeout::default()).await;
+
+    let leader1 = group.get_leader().await.0;
+
+    // 1
```
Comment (Phoenix500526) [inline curp/tests/server_recovery.rs:220]: Same above

File: curp/src/rpc/connect.rs (hunk 1 above)
Comment (markcty) [reply curp/src/rpc/connect.rs:157]: Done.

File: curp/tests/server_recovery.rs (hunk 6 above)
Comment (markcty) [reply curp/tests/server_recovery.rs:220]: Fixed

File: curp/tests/server_election.rs (hunk 2 above)
Comment (markcty) [reply curp/tests/server_election.rs]: Fixed

File: curp/tests/server_election.rs (hunk 3 above)
Comment (markcty) [reply curp/tests/server_election.rs]: Fixed

File: curp/tests/server_election.rs (hunk 4 above)
Comment (markcty) [reply curp/tests/server_election.rs]: Fixed

File: curp/tests/server_recovery.rs (hunk 5 above)
Comment (markcty) [reply curp/tests/server_recovery.rs]: Fixed

**Review by Phoenix500526 — APPROVED** [review]

**Review by themanforfree — APPROVED** [review]

**Pull request merged**